src/datachain/lib/dc.py

```diff
@@ -1225,23 +1224,37 @@ def _effective_signals_schema(self) -> "SignalSchema":
     @overload
     def collect_flatten(self) -> Iterator[tuple[Any, ...]]: ...
 
+    @overload
+    def collect_flatten(self, *, exclude_hidden: bool) -> Iterator[tuple[Any, ...]]: ...
+
     @overload
     def collect_flatten(
         self, *, row_factory: Callable[[list[str], tuple[Any, ...]], _T]
     ) -> Iterator[_T]: ...
 
-    def collect_flatten(self, *, row_factory=None):
+    @overload
+    def collect_flatten(
+        self,
+        *,
+        row_factory: Callable[[list[str], tuple[Any, ...]], _T],
+        exclude_hidden: bool,
+    ) -> Iterator[_T]: ...
+
+    def collect_flatten(self, *, row_factory=None, exclude_hidden: bool = False):
         """Yields flattened rows of values as a tuple.
 
         Args:
             row_factory : A callable to convert row to a custom format.
                           It should accept two arguments: a list of column names and
                           a tuple of row values.
+            exclude_hidden: Whether to exclude hidden signals from the schema.
         """
-        db_signals = self._effective_signals_schema.db_signals()
+        db_signals = self._effective_signals_schema.db_signals(
+            exclude_hidden=exclude_hidden
+        )
         with self._query.ordered_select(*db_signals).as_iterable() as rows:
             if row_factory:
-                rows = (row_factory(db_signals, r) for r in rows)
+                rows = (row_factory(db_signals, r) for r in rows)  # type: ignore[assignment]
```

Comment:
why do we need this type: ignore? can we fix it?

Reply:
Not sure, it was throwing CI error even if the line was not touched. We could come back to fix this but seems like out of scope of this PR.

Reply:
hmm, is CI broken atm? or does it happen only for this PR?

Reply:
It was pre commit hook that was causing the issue which runs only on changed file

Reply:
so, if we run the full mypy on the full code base in main it fails, correct?

Reply:
Not sure, let me try to remove it and push it and see how it behaves in CI.

Reply:
It did not work for some reason. We can come back as a followup but I don't know how to fix this currently honestly